src/lib/operators/jit_operator/operators/jit_aggregate.cpp

```diff
@@ -0,0 +1,312 @@
+        auto& sum_values = context.hashmap.values[sum_column_index].template get_vector<ColumnDataType>();
+        auto count_column_index = column.hashmap_value_2.value().column_index();
+        auto& count_values = context.hashmap.values[count_column_index].get_vector<int64_t>();
+
+        // Then compute the averages.
+        auto avg_values = std::vector<double>(sum_values.size());
+        _compute_averages(sum_values, count_values, avg_values);
+
+        if (column.hashmap_value.is_nullable()) {
+          auto& is_null_values = context.hashmap.values[column.hashmap_value.column_index()].get_is_null_vector();
+          chunk_columns[column.table_position] = std::make_shared<ValueColumn<double>>(avg_values, is_null_values);
+        } else {
+          chunk_columns[column.table_position] = std::make_shared<ValueColumn<double>>(avg_values);
+        }
+      });
+    } else {
+      // All other types of aggregate vectors can be handled just like value columns.
+      resolve_data_type(data_type, [&](auto type) {
+        using ColumnDataType = typename decltype(type)::type;
+        auto column_index = column.hashmap_value.column_index();
+        auto& values = context.hashmap.values[column_index].template get_vector<ColumnDataType>();
+        if (column.hashmap_value.is_nullable()) {
+          auto& is_null_values = context.hashmap.values[column.hashmap_value.column_index()].get_is_null_vector();
+          chunk_columns[column.table_position] = std::make_shared<ValueColumn<ColumnDataType>>(values, is_null_values);
+        } else {
+          chunk_columns[column.table_position] = std::make_shared<ValueColumn<ColumnDataType>>(values);
+        }
+      });
+    }
+  }
+
+  out_table.append_chunk(chunk_columns);
+}
+
+void JitAggregate::add_aggregate_column(const std::string& column_name, const JitTupleValue& value,
+                                        const AggregateFunction function) {
+  auto column_position = _aggregate_columns.size() + _groupby_columns.size();
+
+  switch (function) {
+    case AggregateFunction::Count:
+      // Count aggregates always produce a non-nullable long column.
+      _aggregate_columns.push_back(JitAggregateColumn{column_name, column_position, function, value,
+                                                      JitHashmapValue(DataType::Long, false, _num_hashmap_values++)});
+      break;
+    case AggregateFunction::Sum:
+    case AggregateFunction::Max:
+    case AggregateFunction::Min:
+      Assert(value.data_type() != DataType::String && value.data_type() != DataType::Null,
+             "Invalid data type for aggregate function.");
+      // The data type and nullability of the aggregate depend on the input value.
+      _aggregate_columns.push_back(
+          JitAggregateColumn{column_name, column_position, function, value,
+                             JitHashmapValue(value.data_type(), value.is_nullable(), _num_hashmap_values++)});
```

Comment:
Why are they only nullable on `value.is_nullable()`? Think of `SELECT MAX(a) FROM empty_table`.

Reply:
Fixed.